### DISCUSSION TITLE
RabbitMQ-cluster-operator disappeared from operator index

### DISCUSSION THREAD

**Discussion by ghost** [rabbitmq/cluster-operator#1416] [Q&A] [answered]
Referring to mail sent to rabbitmq-platform email. It looks like the operator has been removed from the registry.redhat.io/redhat/community-operator-index:v4.11 

It exists for v4.10 and v4.12

Relevant PR: https://github.com/redhat-openshift-ecosystem/community-operators-prod/pull/3005

**Comment by DanielePalaia** [comment]
Hi @a164407,

At the moment we have available just an OKD cluster (based on Openshift 4.10)

And as you stated the operator is there in the openshift-marketplace:

```
dpalaia@dpalaiaYMD6R build % oc describe packagemanifest rabbitmq-cluster-operator  -n openshift-marketplace
Name:         rabbitmq-cluster-operator
Namespace:    openshift-marketplace
Labels:       catalog=community-operators
              catalog-namespace=openshift-marketplace
              operatorframework.io/arch.amd64=supported
              operatorframework.io/os.linux=supported
              provider=VMware Tanzu
              provider-url=
Annotations:  <none>
API Version:  packages.operators.coreos.com/v1
Kind:         PackageManifest
Metadata:
  Creation Timestamp:  2022-03-23T10:39:18Z
Spec:
Status:
  Catalog Source:               community-operators
  Catalog Source Display Name:  Community Operators
  Catalog Source Namespace:     openshift-marketplace
  Catalog Source Publisher:     Red Hat
  Channels:
    Current CSV:  rabbitmq-cluster-operator.v2.4.0
```

Is it possible for you to check with the same command in the cluster?

I'm not 100% sure if that registry "registry.redhat.io/redhat/community-operator-index:v4.11" contains just certified operators.

It seems like in my case the catalog is pulled from  "registry.access.redhat.com/redhat/community-operator-index:v4.10"

In any case I think there is no reason why it is just missing for 4.11 :) I can check is something is wrong with the manifest we submitted (but I don't think so) and in case maybe we can try and open an issue here: : https://github.com/redhat-openshift-ecosystem/community-operators-prod/

**Comment by DanielePalaia** [comment]
If this is urgent you may think to install the RabbitMQ operator from the main manifests like described in this github page.

**Comment by ghost** [comment]
Hi @DanielePalaia . For some reason, the package is back in now See below . I can't understand why it was gone for some time. It seems the problem has resolved itself.  

One thing I was wondering about wrt. this situation and operators in general: When you have installed an operator with a certain Subscription and Packagemanifest, can you replace those with another source for your operator without having to delete and reinstall the operator? 

----
opm alpha list packages registry.redhat.io/redhat/community-operator-index:v4.11 | grep rabbit
rabbitmq-cluster-operator                  RabbitMQ-cluster-operator                                         stable
rabbitmq-messaging-topology-operator       rabbitmq-messaging-topology-operator                              stable
rabbitmq-single-active-consumer-operator   rabbitmq-single-active-consumer-operator                          stable

**Comment by DanielePalaia** [comment, marked as answer]
Normally (not using OLM) of course you can update the operators without a delete and reinstall.
Indeed you can kubectl apply -f the new manifest with an udpated operator version.

Now, I didn't really tested directly this situation in OLM. I suppose there shouldn't be issues as long as your catalog source is the same. If you want to change your catalog source I'm not sure if this can be done dynamically. It is more an OLM question, maybe we can have a look in their doc.

In both cases I suggest you to test these scenarios in a test environment to see if everything behave correctly.

For the moment I convert this to discussion as we see it is not a bug.

**Reply by ghost** [reply]
This can be closed now. Not sure why it happened, but you have given me good alternatives to try out if it happens again. Thank you!
